### ISSUE TITLE
Error using var (line 97) The length of W must be compatible with X.

### ISSUE THREAD

**Issue by vld8** [raacampbell/notBoxPlot#8]
Hi Rob, getting an error like Isa's (posted on Matlab site) back in March.  Any insight? Am I missing a prereq toolbox or something? Thanks for your incredibly needed function!

_Error using var (line 97)
The length of W must be compatible with X.

Error in std (line 31)
y = sqrt(var(varargin{:}));

Error in NBP.SEM_calc (line 56)
sem = ( std(vect,'omitnan') ./ sqrt(sum(~isnan(vect))) ) * stdCI ;

Error in notBoxPlot/myPlotter (line 346)
        SEM=intervalFun(Y); %A function handle to a supplied external function

Error in notBoxPlot (line 312)
    [hTemp,statsTemp]=myPlotter(x(f),y(:,f));

Error in notBoxPlot (line 267)
        h(ii)=notBoxPlot(y(f),u(ii),varargin{:},'manualCI',thisCI); %recursive call_

**Comment by raacampbell** [comment]
Can you provide code that reproduces the error? I likely I don't need your actual data but random matrices and/or vectors of the same size. See if you can reproduce it with those and send me the example.

**Comment by vld8** [comment]
Hi Rob,

Thanks for your reply - sorry for my delayed response! I've tried it a few
times, changing the format and organization of the input type to meet the
script criteria (i.e., table columns are y and x; matrix columns are
multiple ys) and get this error with the *var *function call. I am running
matlab R2013b if it helps. Data attached, imported into matlab as table in
this example (2 groups: # 1 and # 3 with unequal sizes, n=25 and n=33).

Thanks for your help!
Val

>> notBoxPlot([DataVizBRIEFinhibit])
Error using var (line 97)
The length of W must be compatible with X.

Error in std (line 31)
y = sqrt(var(varargin{:}));

Error in NBP.SEM_calc (line 56)
sem = ( std(vect,'omitnan') ./ sqrt(sum(~isnan(vect))) ) * stdCI ;

Error in notBoxPlot/myPlotter (line 346)
        SEM=intervalFun(Y); %A function handle to a supplied external
function

Error in notBoxPlot (line 312)
    [hTemp,statsTemp]=myPlotter(x(f),y(:,f));

Error in notBoxPlot (line 267)
        h(ii)=notBoxPlot(y(f),u(ii),varargin{:},'manualCI',thisCI);
%recursive call

**Comment by raacampbell** [comment]
I can't find the attached data. Did you forget to attach them or am I looking in the wrong place?

**Comment by vld8** [comment]
I replied via email - Github didnt save my attachment. My mistake!
[DataViz_BRIEFinhibit.xlsx](https://github.com/raacampbell/notBoxPlot/files/994145/DataViz_BRIEFinhibit.xlsx)

**Comment by raacampbell** [comment]
Can you send the .mat file so that I can replicate your MATLAB command?

**Comment by vld8** [comment]
Hmm. Github doesn't support .mat attachments. I imported the xlsx as a table into my workspace if it helps? Is there another file type I should try to send that would be helpful?
Thank you for your time, Rob!

**Comment by raacampbell** [comment]
In that case, can you post exactly the commands that produce the error? Start at the import step.

**Comment by vld8** [comment]
%% Import the data
data = xlsread('C:\Users\VALERIE\DataViz_BRIEFinhibit.xlsx','BRIEFinhibitP');
%% Create table
DataVizBRIEFinhibit = table;
%% Allocate imported array to column variable names
DataVizBRIEFinhibit.VarName1 = data(:,1);
DataVizBRIEFinhibit.VarName2 = data(:,2);
%% Clear temporary variables
clearvars data raw;
%% call notboxplot function with table as Y input
notBoxPlot([DataVizBRIEFinhibit])


Running this give me error:
_Error using var (line 97)
The length of W must be compatible with X.

Error in std (line 31)
y = sqrt(var(varargin{:}));

Error in NBP.SEM_calc (line 56)
sem = ( std(vect,'omitnan') ./ sqrt(sum(~isnan(vect))) ) * stdCI ;

Error in notBoxPlot/myPlotter (line 346)
        SEM=intervalFun(Y); %A function handle to a supplied external function

Error in notBoxPlot (line 312)
    [hTemp,statsTemp]=myPlotter(x(f),y(:,f));

Error in notBoxPlot (line 267)
        h(ii)=notBoxPlot(y(f),u(ii),varargin{:},'manualCI',thisCI); %recursive call_

**Comment by raacampbell** [comment]
Ok, thanks. I can't reproduce that on R2016b so my hunch is this is an issue with the old version of MATLAB. I think the problem is that your version doesn't have the omitnan flag (I don't remember when that was introduced). 

Can you try editing line 56 of SEM_calc so it now reads:

`sem = ( std(vect) ./ sqrt(sum(~isnan(vect))) ) * stdCI ;`

i.e. remove the `'omitnan'`. See if that fixes it.

**Comment by raacampbell** [comment]
I found an R2013b install and tested. The issue is not specific to your data set: notBoxPlot will fail to run in general on R2013b for the reason mentioned above.

I pushed a fix that will allow you run the function on older MATLAB versions if you have no NaNs. If you do have NaNs it will look for the Stats Toolbox. If you also lack that, it will fail. 

Can you confirm it now works?

**Comment by vld8** [comment]
Thanks for investigating! I copied the new function code (with the NaN fix around line 354) and tried to run with same table input. Still same error (below). Should I have also manually edited line 56 of SEM_calc too? Thanks Rob, Valerie

>> notBoxPlot([DataVizBRIEFinhibit])
Error using var (line 97)
The length of W must be compatible with X.

Error in std (line 31)
y = sqrt(var(varargin{:}));

Error in NBP.SEM_calc (line 56)
sem = ( std(vect,'omitnan') ./ sqrt(sum(~isnan(vect))) ) * stdCI ;

Error in notBoxPlot/myPlotter (line 346)
        SEM=intervalFun(Y); %A function handle to a supplied external function

Error in notBoxPlot (line 312)
    [hTemp,statsTemp]=myPlotter(x(f),y(:,f));

Error in notBoxPlot (line 267)
        h(ii)=notBoxPlot(y(f),u(ii),varargin{:},'manualCI',thisCI); %recursive call
